core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -436,6 +432,7 @@ private boolean add(ResourceInternal<T> resource) {
         return false;
       } else {
         mResources.put(resource.mResource, resource);
+        mCounter.inc();
```

Comment:
These inc and dec lead to paranoia regarding performance, I admit. But I think compared to the cost of making and closing connections within the client, updating a counter should be relatively fine. Also the pooling will make such updates less frequent than it looks on 1st sight.